api/src/pcapi/core/bookings/api.py

```diff
@@ -460,15 +462,29 @@ def cancel_booking_by_offerer(booking: Booking) -> None:
 
 
 def cancel_bookings_from_stock_by_offerer(stock: offers_models.Stock) -> list[Booking]:
-    cancelled_bookings = _cancel_bookings_from_stock(stock, BookingCancellationReasons.OFFERER)
+    cancelled_bookings = _cancel_bookings_from_stock(
+        stock, BookingCancellationReasons.OFFERER, typing.cast(bool, stock.offer.isEvent)
+    )
+    search.async_index_offer_ids([stock.offerId])
+    return cancelled_bookings
+
+
+def cancel_bookings_from_stock_no_longer_free(stock: offers_models.Stock) -> list[Booking]:
+    cancelled_bookings = _cancel_bookings_from_stock(
+        stock,
+        BookingCancellationReasons.OFFERER,
+        stock.offer.subcategoryId in constants.FREE_OFFER_SUBCATEGORY_IDS_TO_ARCHIVE,
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 thought:** ‏le nom de `FREE_OFFER_SUBCATEGORY_IDS_TO_ARCHIVE` ne colle pas.
On a quoi exactement avec ces sous-catégories ? Comment sont elles différentes ?